plugins/create-plugin.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=invalid-name,missing-docstring,consider-using-f-string
+
+import os
+import argparse
+import glob
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+subst = {}
+
+
+def subst_add_string(key: str, value: str) -> None:
+
+    subst[key] = value
+    subst[key.lower()] = value.lower()
+    subst[key.upper()] = value.upper()
+
+
+def subst_replace(data: str) -> str:
+
+    for key, value in subst.items():
+        data = data.replace(key, value)
+    return data
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--vendor", action="store", type=str, default="Hughski", help="Vendor name"
+    )
+    parser.add_argument(
+        "--basename",
+        action="store",
+        type=str,
+        default="Colorhug",
+        help="Plugin basename",
+    )
+    parser.add_argument(
+        "--parent", action="store", type=str, default="Usb", help="Device parent GType"
+    )
+    parser.add_argument(
+        "--year", action="store", type=str, default="2023", help="Copyright year"
+    )
+    parser.add_argument("--author", action="store", type=str, help="Copyright author")
+    parser.add_argument("--email", action="store", type=str, help="Copyright email")
+    args = parser.parse_args()
```

Comment:
I think this will probably need some boundary checking to ensure that certain things are ASCII

Reply:
Agree, done.